sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/ODataDataObject.scala

```diff
@@ -0,0 +1,863 @@
+      dirPath = Option(ioc.newPath(setup.tempFileDirectoryPath.get).resolve(tmpDirName + s"_${ioc.getInstantNow.getEpochSecond}").toString)
+      dirPath.get
+    } else
+      throw new ConfigurationException("TempFileDirectoryPath not configured")
+  }
+
+  /**
+   * Deletes all files in the temporary directory
+   */
+  def clearTempDir(): Unit = {
+    val file = ioc.newFile(this.getDirectoryPath)
+    val files = file.listFiles()
+    files.foreach( f => if (!f.isDirectory) f.delete())
+  }
+
+  /**
+   * Creates the path and the temporary directories if either does not exists.
+   */
+  def makeTempDirIfNotExists(): Unit = {
+    val dirPath = ioc.newPath(this.getDirectoryPath)
+    if (!ioc.fileExists(dirPath)) {
+      ioc.fileCreateDirectories(dirPath)
+    }
+  }
+
+  /**
+   * Write the provided data as a new file into the temporary directory
+   * @param fileName: Name of the file
+   * @param data: Content of the file
+   */
+  def writeToFile(fileName: String, data: String): Unit = {
+    val dirPath : java.nio.file.Path = ioc.newPath(this.getDirectoryPath)
+    val filepath = dirPath.resolve(fileName)
+    val file = ioc.newFile(filepath.toString)
+    val writer = ioc.newBufferedWriter(ioc.newFileWriter(file))
+    writer.write(data)
+    writer.close()
+  }
+
+
+  /**
+   * @return name the current temporary file
+   */
+  def getFileName : String = {
+    val curTime = DateTimeFormatter.ofPattern("yyyyMMdd'_'HHmmss").withZone(ZoneId.of("UTC")).format(ioc.getInstantNow)
+    s"${curTime}_${this.fileCount}.json"
+  }
+
+  /**
+   * Adds the provided response to the buffer
+   * @param response : The response to be added
+   */
+  override def addResponse(response: String): Unit = {
+    initTempDir()
+    writeToFile(getFileName, response)
+    super.addResponse(response)
+    this.fileCount += 1
+  }
+
+  /**
+   * Creates a dataframe which reads all stored temporary files
+   *  @return : The Dataframe referencing the stored responses
+   */
+  override def getDataFrame: DataFrame = {
+    val session = context.sparkSession
+    val dataFrame = session.read.option("wholetext", value = true).text(this.getDirectoryPath).withColumnRenamed("value", "responseString")
+    dataFrame
+  }
+
+  /**
+   * Deletes all temporary stored files.
+   */
+  override def cleanUp(): Unit = {
+    clearTempDir()
+    super.cleanUp()
+  }
+
+  /**
+   * There is no switch path from the FileBuffer. This method returns always the current instance
+   *  @return This instance
+   */
+  override def switchIfNecessary(): ODataResponseBuffer = {
+    this
+  }
+}
+
+class ODataResponseDBFSFileBuffer(tableName: String, setup:ODataResponseBufferSetup, context: ActionPipelineContext, ioc: ODataIOC) extends ODataResponseBuffer(setup, context) {
+
+  private var dirInitialized : Boolean = false
+  private implicit val filesystem: FileSystem = ioc.newHadoopFsWithConf(ioc.newHadoopPath("dbfs://"), context)
```

Comment:
I dont like that this is dbfs specific... we should be able to freely configure the path, so this can be used also outside of Databricks (also dbfs is unsafe... a better choice might be a node local directory)

Reply:
I dont like it either but I saw no other way. I've prepared two file buffer classes, one that handles regular file systems and one that handles DBFS. For my understanding, I have to use DBFS when running in DataBricks and to be able to support a larger data volume.